pythran/pythonic/types/complex.hpp

```diff
@@ -63,4 +63,38 @@ namespace pythonic
   }
 }
 
+#ifdef ENABLE_PYTHON_MODULE
+
+#include "pythonic/python/core.hpp"
+
+namespace pythonic
+{
+
+  template <class T>
+  struct to_python<std::complex<T>> {
+    static PyObject *convert(std::complex<T> const &c)
+    {
+      return PyComplex_FromDoubles(c.real(), c.imag());
+    }
+  };
+
+  template <class T>
+  struct from_python<std::complex<T>> {
+    static bool is_convertible(PyObject *obj)
+    {
+      return PyComplex_Check(obj) or PyFloat_Check(obj) or PyInt_Check(obj);
+    }
+    static std::complex<T> convert(PyObject *obj)
+    {
+      if (PyComplex_Check(obj))
+        return {PyComplex_RealAsDouble(obj), PyComplex_ImagAsDouble(obj)};
+      else if (PyFloat_Check(obj))
+        return {PyFloat_AsDouble(obj), 0.};
+      else
+        return {(double)PyInt_AsLong(obj), 0.};
```

Comment:
> Here you check type twice. Once in `is_convertible` and another in `convert`. Why not using only `convert` and return nullptr when it fail?

I agree with the problem of the double check. But Your option would
imply either unnecessary conversions, or convoluted code, as in:

```
// we want to try two functions : f(set<int>, set<int>) or f (list<int>, list<int>)

if(set<int> arg0 = try_convert<set<int>>(py_arg0)) {
    if(set<int> arg1 = try_convert<set<int>>(py_arg1)) {
        return f0(arg0, arg1);
    }
    // arg0 will be destroyed here, so that's a potentially very costly
    // conversion that proved to be useless
}
else if (list<int> arg0 = try_convert<list<int>>(py_arg0)) {
    if(list<int> arg1 = try_convert<list<int>>(py_arg1)) {
        return f1(arg0, arg1);
    }  
    // arg0 will be destroyed here too
}
```